### PR TITLE
Dogs table

### PR DESCRIPTION
**Pull request by JustinSteel**
## Description

Created dog table, with all crud functions, and testing for said crud.  

## Type of change
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Crud 

- [x]  fully tests happy and sad paths

## Checklist

- [x] code has been self reviewed
- [x] code runs without any errors
- [x] all tests pass

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,7 +1,9 @@
 Rails.application.routes.draw do
   namespace :api do
     namespace :v1 do
-      resources :users
+      resources :users do
+        resources :dogs
+      end
```
Comment (ttakahashi1591) [inline config/routes.rb:6]: This name-spacing looks great here as a `User` can have many `Dogs`.

File: db/migrate/20240105172205_create_dogs.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class CreateDogs < ActiveRecord::Migration[7.1]
+  def change
+    create_table :dogs do |t|
+      t.string :name
+      t.string :breed
+      t.integer :age
+      t.integer :size, default: 0
```
Comment (ttakahashi1591) [inline db/migrate/20240105172205_create_dogs.rb:7]: I like the use of the enum to support with `Dog Size`.

File: spec/requests/api/v1/dogs_request_spec.rb [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        neutered: true
+      })
+      headers = {"CONTENT_TYPE" => "application/json"}
+  
+      post api_v1_user_dogs_path(user.id, dog_params), headers: headers, params: JSON.generate(dog: dog_params)
+
+      expect(response).to_not be_successful 
+      expect(response).to have_http_status(401)
+    end
+  end
```
Comment (ttakahashi1591) [inline spec/requests/api/v1/dogs_request_spec.rb:116]: Nice job with including the sad path testing here.

**Review by delaneymiranda1 — APPROVED** [review]
Great work. We may want to change the naming for the spay/neuter attribute to 'fixed' since neuter is specific to male. That's what i was using on FE.. but that's something we can discuss in our next meeting if necessary.

**Pull request merged**